tribe-customizations/v2/week/grid-body.php

```diff
@@ -0,0 +1,111 @@
+ *                                 `[ <Y-m-d> => [ 'events' => [ ...$multiday_events], 'more_events' => <int> ] ]`.
+ * @var bool  $has_multiday_events Boolean whether the week has multiday events or not.
+ * @var array $events              An array of each day non multi-day events, if any, in the shape `[ <Y-m-d> => [ ...$events ] ]`.
+ * @var array $ext_options         An array of the extension settings.
+ * @var int   $count               int to count which day of the week we are on.
+ */
+
+/**
+ * Get the settings
+ */
+$ext_options['grid_start_time'] = tribe_get_option( 'tribe_ext_limit_week_view_time_range_start_time', 0 );
+$ext_options['grid_end_time']   = tribe_get_option( 'tribe_ext_limit_week_view_time_range_end_time', 24 );
+$ext_options['show_grid']       = tribe_get_option( 'tribe_ext_limit_week_view_time_range_show_grid', false );
+
+/**
+ * Making sure that the values are within the accepted range and are integer,
+ * otherwise set the default value.
+ */
+if (
+	$ext_options['grid_start_time'] < 0
+	|| $ext_options['grid_start_time'] > 23
+) {
+	$ext_options['grid_start_time'] = 0;
+}
+
+if (
+	$ext_options['grid_end_time'] < 1
+	|| $ext_options['grid_end_time'] > 24
+) {
+	$ext_options['grid_end_time'] = 24;
+}
+?>
+
+<div class="tribe-events-pro-week-grid__body" role="rowgroup">
+
+	<?php if ( count( $multiday_events ) && $has_multiday_events ) : ?>
+
+		<div class="tribe-events-pro-week-grid__multiday-events-row-outer-wrapper">
+			<div class="tribe-events-pro-week-grid__multiday-events-row-wrapper">
+				<div
+					class="tribe-events-pro-week-grid__multiday-events-row"
+					data-js="tribe-events-pro-week-multiday-events-row"
+					role="row"
+				>
+
+					<?php $this->template( 'week/grid-body/multiday-events-row-header' ); ?>
+
+					<?php
+					foreach ( $multiday_events as $day => list( $day_multiday_events, $more_events ) ) :
+						$this->template(
+							'week/grid-body/multiday-events-day',
+							[
+								'day'         => $day,
+								'events'      => $day_multiday_events,
+								'more_events' => $more_events,
+							]
+						);
+					endforeach;
+					?>
+
+				</div>
+			</div>
+		</div>
+
+	<?php endif; ?>
+
+	<div class="tribe-events-pro-week-grid__events-scroll-wrapper">
+		<div class="tribe-events-pro-week-grid__events-row-outer-wrapper" data-js="tribe-events-pro-week-grid-events-row-outer-wrapper">
+			<div class="tribe-events-pro-week-grid__events-row-wrapper" data-js="tribe-events-pro-week-grid-events-row-wrapper">
+				<div class="tribe-events-pro-week-grid__events-row" role="row">
+
+					<?php $this->template( 'week/grid-body/events-row-header', [ 'ext_options' => $ext_options ] ); ?>
+
+					<?php
+					$count = 0;
+					foreach ( $events as $day => $day_events ) :
+						$this->template(
+							'week/grid-body/events-day',
+							[
+								'events'      => $day_events,
+								'ext_options' => $ext_options,
+								'count'       => $count,
+							]
+						);
+						$count ++;
```

Comment:
This can actually move up to line 99 above `'count' => $count++,`

But it's fine the way it is - and probably a bit less confusing.

If you are interested: http://sandbox.onlinephpfunctions.com/code/61f1ff440462c541049849cf4c7370928f946d28

Reply:
Ah, that's great, thanks for sharing. Didn't know you could do that.

For easier understanding I would leave it like that for now.